### PR TITLE
Add rebuild support when setting a child online/offline

### PR DESCRIPTION
**Pull request by paulyoong**
- Setting a child online kicks off a rebuild job for the destination child
- Setting a child offline stops any existing rebuilds for the destination child

Additional changes:
- Fix the "child_operation" call to support online and offline operations
- Make the "stop_rebuild" call idempotent
- Fix the "online" and "offline" rpc calls so that they do not return a result

### CODE REVIEW

File: csi/src/mayastor_svc.rs [hunk 1]
```diff
@@ -307,8 +307,21 @@ impl service::mayastor_server::Mayastor for MayastorService {
     ) -> Result<Response<Null>, Status> {
         let msg = request.into_inner();
         trace!("{:?}", msg);
-        jsonrpc::call::<_, ()>(&self.socket, "offline_child", Some(msg))
-            .await?;
+        match msg.action {
```
Comment (tiagolobocastro) [inline csi/src/mayastor_svc.rs]: Could we add a helper child action to string? And maybe we return an error is it's an invalid action?

File: mayastor/src/bdev/nexus/nexus_rpc.rs [hunk 2]
```diff
@@ -167,21 +167,29 @@ pub(crate) fn register_rpc_methods() {
         fut.boxed_local()
     });
 
-    jsonrpc_register("offline_child", |args: ChildNexusRequest| {
-        let fut = async move {
-            let nexus = nexus_lookup(&args.uuid)?;
-            nexus.offline_child(&args.uri).await
```
Comment (tiagolobocastro) [inline mayastor/src/bdev/nexus/nexus_rpc.rs:170]: why are we not returning the state now?

**Review by tiagolobocastro — CHANGES_REQUESTED** [review]

File: mayastor/src/bdev/nexus/nexus_rpc.rs (hunk 2 above)
Comment (paulyoong) [reply mayastor/src/bdev/nexus/nexus_rpc.rs:170]: When calling the offline/online rpc methods, the rpc clients were complaining that they were receiving a return value (the nexus state) when they weren't expecting to get anything back (they were expecting a unit type back which is () or empty). If you look at the child_operation rpc implementation in mayastor_svc.rs we were doing something similar i.e. ignoring the returned value and returning null.

**Review by tiagolobocastro — DISMISSED** [review]

**Review by tiagolobocastro — APPROVED** [review]
reviewed the new changes on a call so LGTM

**Pull request closed without merging**